app/time_tracking/work_statistic/views.py

```diff
@@ -0,0 +1,148 @@
+
+
+@api_view(['GET'])
+@permission_classes((IsAuthenticated, ))
+def year_statistics_detail(request, format=None):
+    """
+    Retrieve work time since a 356 days ago.
+    """
+    if request.method == 'GET':
+        work_times = WorkTime.objects.filter(
+            owner=request.user,
+            work_time__isnull=False,
+            end_time__gt=timezone.now() - timedelta(days=356)
+        ).aggregate(Sum('work_time'))
+        responce = {
+            "total_working_time_in_seconds": work_times['work_time__sum']
+        }
+
+        return Response(responce, status.HTTP_200_OK)
+
+    return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+
+@api_view(['GET'])
+@permission_classes((IsAuthenticated, ))
+def employees_arrival_and_leving_time_statistics_detail(request, format=None):
+    """
+    Retrieve the employees arrival time and leaving times.
+    """
+    if request.method == 'GET':
+        responce = []
+        for user in User.objects.filter(is_staff=False):
+            avg_work_time_starts = WorkTime.objects.filter(owner=user).aggregate(Avg('start_unix_time'))['start_unix_time__avg']
+            avg_work_time_ends = WorkTime.objects.filter(owner=user).aggregate(Avg('end_unix_time'))['end_unix_time__avg']
+
+            if avg_work_time_starts is not None:
+                avg_work_time_starts = datetime.fromtimestamp(avg_work_time_starts)
+            if avg_work_time_ends is not None:
+                avg_work_time_ends = datetime.fromtimestamp(avg_work_time_ends)
+
+            responce.append({
+                'user': user.username,
+                'avarage_arraival_time': avg_work_time_starts,
+                'avarage_leaving_time': avg_work_time_ends,
+            })
+
+        return Response(responce, status.HTTP_200_OK)
+
+    return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+
+@api_view(['GET'])
+@permission_classes((IsAuthenticated, ))
+def working_ours_to_leaving_hours_detail(request, format=None):
+    """
+    Retrieve the team working hours to leaving hours
+    """
+    if request.method == 'GET':
+        working_hours = 0
+        leaving_hours = 0
+        for user in User.objects.filter():
+            work_times_grouped_by_owner_and_date = WorkTime.objects.filter(
+                owner=user,
+                end_unix_time__isnull=False,
+                start_unix_time__isnull=False
+            ).values(
+                'days_count', 'start_unix_time', 'end_unix_time'
+            ).annotate(days=Count('days_count')).order_by('-start_unix_time')
+
+            length = len(work_times_grouped_by_owner_and_date)
+            index = 0
+
+            for work_time in work_times_grouped_by_owner_and_date:
+                working_hours += work_time['end_unix_time'] - work_time['start_unix_time']
+                if ++index < length - 1:
+                    leaving_hours += work_times_grouped_by_owner_and_date[index]['start_unix_time'] - work_time['end_unix_time']
```

Comment:
this is completely inaccurate, it should be grouped by day
For example, Ayman worked in Aug 11, 2 hours in the morning and 2 hours in the afternoon

[aug 11]
working_hours = 4
leaving_hours = 8 (total working hours per day) - working_hours = 4
work_on_leave_time = working_hours / leaving_hours * 100

Reply:
Ooh, I see, that is different from what I was trying to achieve,
on the emails, Waseem gave me this example:
> Users can checkin for example at 8am then checkout at 12pm then checkin again at 3pm and checkout at 6pm. In this example, user worked for a total of 7 hours during that day and left for 3 hours, as simple as that.

so I was trying to calculate the working hours for each day and the leaving time between each work time on the same day.
Using total working hours per day is actually easier I'll fix it.

Reply:
@Moath-Zaghdad thanks for explaining this, actually the mentioned example above is not different from what I wrote. 

> so I was trying to calculate the working hours for each day and the leaving time between each work time on the same day
Your code doesn't group entries by day, can you explain more about this?

Reply:
I can't explain the code itself because as you said it's not correct.
This was the last piece of code that I wrote, and I knew there was something fishy about it, It was last minute gambling, I was either going to add it or leave the feature unimplemented, and I decided, unwisely, to commit the code.

Sorry that I'm overcomplicating this, but I actually see these two examples differently:
One suggests that if I know the sum of the working hours a day for an employee I can subtract 8 from it to result with the total leaving hours.
The other example suggests that leaving hours doesn't correlate with the Total-Working-Hours-A-Day but with the total hours between each check-outs and check-ins per day.

If you could please explain this example for me it would be a great help,
Let's say that the `Total-Working-Hours-A-Day` is **8**
| Check-in Time  | Check-out Time     | Work Time     |
| :-------------: | :----------: | :-----------: |
|  08:00 | 11:00   |  3 Hs   |
|  12:00 | 16:00   |  4 Hs   |
|  18:00 | 19:00   |  1 Hs   |

The employee did work for a full 8 hours.
On my old understanding of the leaving hours, the employee left for 3 hours that day.
And my understanding of the last explanation made me decide that the employee leaving time is Zero on this case.

Reply:
@awartani  @Moath-Zaghdad Waseem and Ayman proposed a different definition for leaving hours, both of them are sensible, to clear out any confusion here, we should pick one of them,  let's stick with Waseem approach since it the first proposed approach. so that the leaving hours is the difference between each check-in and check-out **in the same day**.

So, if the employee has the following events:
Day | Check-in Time | Check-out Time | Work Time | Leave Time
-- | -- | -- | -- | -- 
12/8/2020 | 08:00 | 11:00 | 3 Hs | 0Hs
12/8/2020 | 12:00 | 16:00 | 4 Hs | 1Hs
12/8/2020 | 18:00 | 19:00 | 1 Hs | 2Hs
13/8/2020 | 11:00 | 15:00 | 4 Hs | 0Hs

Then:
On `12/8/2020`, working-time=8h and leaving-time=3h
On `13/8/2020`, working-time=4h and leaving-time=0h

@Moath-Zaghdad can you update the code to match this logic? if you are already implemented Ayman approach that's fine, but let us know.

Reply:
Thanks for the clarification, I'll fix my code based on to this logic.
No, I have not implemented Aymans' approach.

Reply:
I refactored this, and I'll write some tests for it.